eng/pipelines/templates/stages/archetype-sdk-client.yml

```diff
@@ -73,13 +73,16 @@ parameters:
 - name: PostTestSteps
   type: stepList
   default: []
-
+- name: CMakeGenerationTimeoutInMinutes 
+  type: number
+  default: 120
 
 stages:
   - stage: CMakeGeneration
     jobs:
       - template: /eng/pipelines/templates/jobs/cmake-generate-jobs.yml
         parameters:
+          TimeoutInMinutes: ${{ parameters.CMakeGenerationTimeoutInMinutes }}
```

Comment:
I am not very familiar with how each of the CI stage is laid out in yaml, so trying to understand this a bit.
In this stage and jobs, the `TimeoutInMinutes` is in the parameters list, but it is using the `cmake-generate-jobs.yml` as a template, which has a `timeoutInMinutes` as a jobs property. Why do we need to have 2 timeout variables set, both `CMakeGenerationTimeoutInMinutes` and `TimeoutInMinutes`?

Reply:
the CMakeGenerationTimeoutInMinutes is a parameter for the script , with a default value of 120. there is another timeout defined LiveTestTimeout with it's default value. I did not want to reuse the LiveTestTimeout as it has another purpose. TimoutInMinutes is a parameter for the next template down the line which does not define any other timeouts thus i could be the first to use timeout in minutes.